### PR TITLE
Small Deneb cleanups

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Pass `SpecConfigDeneb` in MessageHandlers
- No need to overwrite `getMaxRequestBlobSidecars` in `SpecConfigDeneb` since it is available in the config
- Remove unnecessary consistency check in CKZG4844. It is covered by the library.

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java [hunk 1]
```diff
@@ -132,13 +132,6 @@ public boolean verifyBlobKzgProofBatch(
       final List<KZGCommitment> kzgCommitments,
       final List<KZGProof> kzgProofs)
       throws KZGException {
-    if (blobs.size() != kzgCommitments.size() || blobs.size() != kzgProofs.size()) {
```
Comment (mehdi-aouadi) [inline infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java:135]: Don't we need this check anymore?

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java [hunk 2]
```diff
@@ -54,19 +52,19 @@ public class BlobSidecarsByRootMessageHandler
   private static final Logger LOG = LogManager.getLogger();
 
   private final Spec spec;
-  private final UInt64 denebForkEpoch;
+  private final SpecConfigDeneb specConfig;
```
Comment (mehdi-aouadi) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java]: I'd name it `specConfigDeneb` (Same int the by range message handler)

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java (hunk 1 above)
Comment (mehdi-aouadi) [reply infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java:135]: Oh I see you edited the PR description :) Check done in the library... Got it

**Review by mehdi-aouadi — APPROVED** [review]
LGTM

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java (hunk 2 above)
Comment (StefanBratanov) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootMessageHandler.java]: good catch thanks

**Pull request merged**